### PR TITLE
docs: Add JOSS paper to preferred citation and DOI Badge

### PR DESCRIPTION
**Pull request by kratsg**
# Pull Request Description

JOSS is out! https://twitter.com/JOSS_TheOJ/status/1357410376124669952

[DOI](https://doi.org/10.21105/joss.02823)

ReadTheDocs build: https://pyhf.readthedocs.io/en/docs-addjossbadge/

# Checklist Before Requesting Reviewer

- [x] Tests are passing
- [x] "WIP" removed from the title of the pull request
- [x] Selected an Assignee for the PR to be responsible for the log summary

# Before Merging

For the PR Assignees:

- [x] Summarize commit messages into a comprehensive review of the PR
```
* Add JOSS paper BibTeX entry to preferred citation
* Add JOSS DOI badge to README
   - c.f. https://doi.org/10.21105/joss.02823
* Add test for JOSS paper in pyhf --cite output
```

### CODE REVIEW

**Review by matthewfeickert — CHANGES_REQUESTED** [review]
@kratsg this is good but we should also update the citations sections to have people cite both the paper and the Zenodo DOI.

**Comment by kratsg** [comment]
> @kratsg this is good but we should also update the citations sections to have people cite both the paper and the Zenodo DOI.

done

File: src/pyhf/data/citation.bib [hunk 1]
```diff
@@ -5,3 +5,16 @@ @software{pyhf
   doi = {10.5281/zenodo.1169739},
   url = {https://github.com/scikit-hep/pyhf},
 }
+
+@article{pyhf_joss,
```
Comment (matthewfeickert) [inline src/pyhf/data/citation.bib:9]: This should also appear in the README citation section with a clear request to cite both.

**Review by matthewfeickert — CHANGES_REQUESTED** [review]
We should also update the language in the README and `docs/citations.rst`

**Comment by matthewfeickert** [comment]
> We should also update the language in the README and `docs/citations.rst`

So for both maybe something like

> The preferred BibTeX entry for citation of ``pyhf`` includes both the JOSS paper and the Zenodo archive

**Comment by kratsg** [comment]
> So for both maybe something like

done

File: README.rst [hunk 2]
```diff
@@ -275,9 +275,10 @@ announcements you can join the |pyhf-announcements mailing list|_.
 Citation
 --------
 
-As noted in `Use and
-Citations <https://scikit-hep.org/pyhf/citations.html>`__, the preferred
-BibTeX entry for citation of ``pyhf`` is
+As noted in `Use and Citations <https://scikit-hep.org/pyhf/citations.html>`__,
+the preferred BibTeX entries for citation of ``pyhf`` includes both the `JOSS
+<https://joss.theoj.org/>`__ paper and the `Zenodo <https://zenodo.org/>`__
+archive:
```
Comment (matthewfeickert) [inline README.rst:281]:
```suggestion
As noted in `Use and Citations <https://scikit-hep.org/pyhf/citations.html>`__,
the preferred BibTeX entries for citation of ``pyhf`` includes both the
`Zenodo <https://zenodo.org/>`__ archive and the
`JOSS <https://joss.theoj.org/>`__ paper:
```

As the render on the citations page shows the Zenodo entry first can we have the language match that?

contradict_myself

File: docs/citations.rst [hunk 3]
```diff
@@ -1,10 +1,16 @@
...
+
+   <p id="dev-version"><strong>Warning:</strong> This is a development version and should not be cited. The latest stable version is at <a href="https://pyhf.readthedocs.io/">ReadTheDocs</a>.</p>
+
 Citation
 --------
 
-The preferred BibTeX entry for citation of ``pyhf`` is
+The preferred BibTeX entries for citation of ``pyhf`` includes both the `JOSS
+<https://joss.theoj.org/>`__ paper and the `Zenodo <https://zenodo.org/>`__
+archive:
```
Comment (matthewfeickert) [inline docs/citations.rst:13]:
```suggestion
The preferred BibTeX entries for citation of ``pyhf`` includes both the  `Zenodo <https://zenodo.org/>`__
and the `JOSS <https://joss.theoj.org/>`__ paper:
```

Same thought as the above

**Review by matthewfeickert — COMMENTED** [review]
@kratsg Thought on ordering of the text. Not sure of your feelings here.

**Comment by kratsg** [comment]
> Thought on ordering of the text. Not sure of your feelings here.

happy to add :) thanks.

File: docs/citations.rst [hunk 4]
```diff
@@ -1,10 +1,15 @@
...
+.. raw:: html
+
+   <p id="dev-version"><strong>Warning:</strong> This is a development version and should not be cited. To find the specific version to cite, please go to <a href="https://pyhf.readthedocs.io/">ReadTheDocs</a>.</p>
+
 Citation
 --------
 
-The preferred BibTeX entry for citation of ``pyhf`` is
+The preferred BibTeX entries for citation of ``pyhf`` includes both the  `Zenodo <https://zenodo.org/>`__
+and the `JOSS <https://joss.theoj.org/>`__ paper:
```
Comment (matthewfeickert) [inline docs/citations.rst:12]:
```suggestion
archive and the `JOSS <https://joss.theoj.org/>`__ paper:
```

**Review by matthewfeickert — CHANGES_REQUESTED** [review]
"archive" got dropped along the way.

**Comment by matthewfeickert** [comment]
@kratsg the tests are failing on doctest now

```
=================================== FAILURES ===================================
________________________ [doctest] pyhf.utils.citation _________________________
138 
139     Get the bibtex citation for pyhf
140 
141     Example:
142 
143         >>> import pyhf
144         >>> pyhf.utils.citation(True)
Expected:
    '@software{pyhf,  author = "{Heinrich, Lukas and Feickert, Matthew and Stark, Giordon}",  title = "{pyhf: v0.5.4}",  version = {0.5.4},  doi = {10.5281/zenodo.1169739},  url = {https://github.com/scikit-hep/pyhf},}'
Got:
    '@software{pyhf,  author = "{Heinrich, Lukas and Feickert, Matthew and Stark, Giordon}",  title = "{pyhf: v0.5.4}",  version = {0.5.4},  doi = {10.5281/zenodo.1169739},  url = {https://github.com/scikit-hep/pyhf},}@article{pyhf_joss,  doi = {10.21105/joss.02823},  url = {https://doi.org/10.21105/joss.02823},  year = {2021},  publisher = {The Open Journal},  volume = {6},  number = {58},  pages = {2823},  author = {Lukas Heinrich and Matthew Feickert and Giordon Stark and Kyle Cranmer},  title = {pyhf: pure-Python implementation of HistFactory statistical models},  journal = {Journal of Open Source Software}}'
```

File: README.rst [hunk 5]
```diff
@@ -275,9 +275,10 @@ announcements you can join the |pyhf-announcements mailing list|_.
 Citation
 --------
 
-As noted in `Use and
-Citations <https://scikit-hep.org/pyhf/citations.html>`__, the preferred
-BibTeX entry for citation of ``pyhf`` is
+As noted in `Use and Citations <https://scikit-hep.org/pyhf/citations.html>`__,
+the preferred BibTeX entries for citation of ``pyhf`` includes both the
```
Comment (matthewfeickert) [inline README.rst:279]:
```suggestion
the preferred BibTeX entry for citation of ``pyhf`` includes both the
```

I can't English well today it seems. :(

File: docs/citations.rst [hunk 6]
```diff
@@ -1,10 +1,15 @@
...
 
+.. raw:: html
+
+   <p id="dev-version"><strong>Warning:</strong> This is a development version and should not be cited. To find the specific version to cite, please go to <a href="https://pyhf.readthedocs.io/">ReadTheDocs</a>.</p>
+
 Citation
 --------
 
-The preferred BibTeX entry for citation of ``pyhf`` is
+The preferred BibTeX entries for citation of ``pyhf`` includes both the  `Zenodo <https://zenodo.org/>`__
```
Comment (matthewfeickert) [inline docs/citations.rst:11]:
```suggestion
The preferred BibTeX entry for citation of ``pyhf`` includes both the  `Zenodo <https://zenodo.org/>`__
```

**Review by matthewfeickert — APPROVED** [review]

**Comment by matthewfeickert** [comment]
@kratsg sorry this one took so long :sweat_smile: but this LGTM. :+1:

**Pull request merged**